proposed/2021/NuGet.orgTFMs.md

```diff
@@ -0,0 +1,96 @@
+# NuGet Target Frameworks on Package Details Page 
+
+- Author Name (https://github.com/jcjiang)
+- Start Date (2021-02-16)
+- GitHub Issue (https://github.com/NuGet/NuGetGallery/issues/8387)
+- GitHub PR ()
+
+## Summary
+
+When a developer opens the package details page or tab on NuGet.org or NuGet rooling, they will see what frameworks the package can support. 
+
+## Motivation
+
+### Problem: What problem is this solving? 
+
+Developers cannot use the package details page to determine what frameworks a NuGet package can support. In 2021, we live in a world where there are three evolutions of .NET coexisting on NuGet; .NET Framework, .NET Core, and .NET 5.  
+
+Developers should be able to determine whether a package works for their project without having to first download it or rely on the author’s documentation. 
+
+### Why: How do we know this is a real problem and worth solving? 
+
+Finding a package that is compatible with your project is a challenge for the average developer in the .NET ecosystem. Over 21% of developers fail to install a package today. One of the top NuGet errors issued in telemetry is [NU1202](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1202), which shows that we allow developers to install packages that are not compatible with their project or solution’s target framework. 
+
+### Success: How do we know if we’ve solved this problem? 
+
+- By measuring the change in the percentage of successful installs and lowering the amount of [NU1202](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1202) errors in telemetry. 
+- By measuring the usage of target framework & supported platform filters on NuGet.org & Visual Studio. 
+- By gauging excitement/disappointment after blogging about this feature & analyzing sentiment on Twitter/GitHub/DevCom/etc. 
+
+### Audience: Who are we building for? 
+
+We are building this for the .NET developer who is browsing & making decisions about including a NuGet package into their project based on target framework & supported platform criteria. 
+
+## Explanation
+
+### Functional Explanation
+
+#### Context and Scope 
+When a developer accesses the details page for a NuGet package, they will see information on target frameworks in a dropdown menu. The framework information will be surfaced from the package. In cases where versions are not specified (such as .net) or other unexpected scenarios, ‘Version Not Specified’ or similar messages will be surfaced. 
+
+##### Minimal Requirements 
+- List of Frameworks on NuGet.org
```

Comment:
I don't know if there are any differences between PackageReference and packages.config compatibility. Client team will know for sure. If there is, we should note that this is _PackageReference_ compatibiliy

Reply:
@nkolev92 or @zivkan have any thoughts here?

Reply:
There are some differences, yes.

For example, packages.config will accept a package with `lib/foo.dll`, and this package will be deemed compatible with all TFMs. But PackageReference will completely ignore it, and only accept packages with `lib\<tfm>\foo.dll` (assuming the package doesn't have other assets.

I haven't yet read this spec, so I don't know if my comment makes sense in whatever context this section of the document is, but it's absolutely possible for a package to work only for packages.config, only for PackageReference, or work for both.

Reply:
You comment makes sense in whatever context this is 🥇

Reply:
This is pertinent to our TFM-assessment logic on package push. If we're looking to support `PackageReference`, we need to not fall back to nuspec reads when the package doesn't contain lib/<tfm>/foo.dll, and there's definitely no case for inspecting dlls for target framework metadata.

Reply:
Added `- Support for PackageReference project restore compatibility only.`